samples/analyze-streaming-automl-action-recognition.js

```diff
@@ -0,0 +1,93 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+async function main(
+  path = 'YOUR_LOCAL_FILE',
+  projectId = 'YOUR_GCP_PROJECT',
+  modelId = 'YOUR_AUTOML_MODELID'
+) {
+  // [START video_streaming_automl_action_recognition_beta]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const path = 'Local file to analyze, e.g. ./my-file.mp4';
+  // const modelId = 'Vertex model'
+  // const projectId = 'Your GCP Project'
+
+  const {StreamingVideoIntelligenceServiceClient} =
+    require('@google-cloud/video-intelligence').v1p3beta1;
+  const fs = require('fs');
+
+  // Instantiates a client
+  const client = new StreamingVideoIntelligenceServiceClient();
+
+  // Streaming configuration
+  const modelName = `projects/${projectId}/locations/us-central1/models/${modelId}`;
+  const configRequest = {
+    videoConfig: {
+      feature: 'STREAMING_AUTOML_ACTION_RECOGNITION',
+        automlActionRecognitionConfig: {
+        modelName: modelName,
+      },
+    },
+  };
+
+  const readStream = fs.createReadStream(path, {
+    highWaterMark: 5 * 1024 * 1024, //chunk size set to 5MB (recommended less than 10MB)
+    encoding: 'base64',
+  });
+  //Load file content
+  // Note: Input videos must have supported video codecs. See
+  // https://cloud.google.com/video-intelligence/docs/streaming/streaming#supported_video_codecs
+  // for more details.
+  const chunks = [];
+  readStream
```

Comment:
As written, this is going to buffer all of the contents from the file you're reading into memory.  Instead of fully buffering like this, could we start off by calling `stream.write(configRequest);` and then inside of the `.on('data')` handler, write chunks to the stream instead of adding them to the buffer?

Reply:
Alternatively - this is a pretty classic use case for a `Transform` stream!  It could be worth playing with something like this:

```js

const transformStream = new Transform({
  transform(chunk, encoding, callback) {
    callback(null, {
        inputContent: chunk.toString(),
      });
  }
});
readStream.pipe(transformStream).pipe(stream);
```

Reply:
@bcoe I'm trash at streams, please make sure I'm not making this up

Reply:
I'm so so with streams too, I think a transform stream should work, but you'd want to be careful that streams have two modes, [object mode](https://nodejs.org/api/stream.html#stream_object_mode) and the standard Buffer mode.

This seems like it's loading the file as a buffer, but then wanting to transform it into an object output for each chunk?